src/Http/Http.Results/src/JsonHttpResultOfT.cs

```diff
@@ -13,38 +16,53 @@ namespace Microsoft.AspNetCore.Http.HttpResults;
 /// </summary>
 public sealed partial class JsonHttpResult<TValue> : IResult, IStatusCodeHttpResult, IValueHttpResult, IValueHttpResult<TValue>, IContentTypeHttpResult
 {
-    /// <summary>
-    /// Initializes a new instance of the <see cref="Json"/> class with the values.
-    /// </summary>
-    /// <param name="value">The value to format in the entity body.</param>
-    /// <param name="jsonSerializerOptions">The serializer settings.</param>
-    internal JsonHttpResult(TValue? value, JsonSerializerOptions? jsonSerializerOptions)
-        : this(value, statusCode: null, contentType: null, jsonSerializerOptions: jsonSerializerOptions)
-    {
-    }
+    private readonly JsonTypeInfo? _jsonTypeInfo;
 
     /// <summary>
     /// Initializes a new instance of the <see cref="Json"/> class with the values.
     /// </summary>
     /// <param name="value">The value to format in the entity body.</param>
-    /// <param name="statusCode">The HTTP status code of the response.</param>
     /// <param name="jsonSerializerOptions">The serializer settings.</param>
-    internal JsonHttpResult(TValue? value, int? statusCode, JsonSerializerOptions? jsonSerializerOptions)
-        : this(value, statusCode: statusCode, contentType: null, jsonSerializerOptions: jsonSerializerOptions)
+    /// <param name="statusCode">The HTTP status code of the response.</param>
+    /// <param name="contentType">The value for the <c>Content-Type</c> header</param>
+    [RequiresDynamicCode(JsonHttpResultTrimmerWarning.SerializationRequiresDynamicCodeMessage)]
+    [RequiresUnreferencedCode(JsonHttpResultTrimmerWarning.SerializationUnreferencedCodeMessage)]
```

Comment:
Remind me why this is unsafe after the other work we did to make the underlying JSON APIs that these call into safe?

Reply:
This is unsafe because it is the scenario where the user provides the options, could be resolved by DI or not, so, we need to check and initialize for reflection.